app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java

```diff
@@ -0,0 +1,29 @@
+package ingsw.group1.findmyphone.contacts;
+
+import android.widget.SearchView;
+
+/**
+ * Class implements {@link SearchView.OnQueryTextListener}.
+ * This is a listener to activate the search on the contact list through a {@link SMSContactRecyclerAdapter}.
+ *
+ * @author Giorgia Bortoletti
+ */
+public class SMSContactSearchListener implements SearchView.OnQueryTextListener {
+
+    private SMSContactRecyclerAdapter recyclerAdapter;
+
+    public SMSContactSearchListener(SMSContactRecyclerAdapter recyclerAdapter) {
+        this.recyclerAdapter = recyclerAdapter;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        recyclerAdapter.getFilter().filter(newText);
+        return false;
```

Comment:
Why does this returns `false`? Also, the other methods lack documentation.

Reply:
I'll add the specs, thanks.
It is the same return true or false because this method calls the object to filter the data. They are immediately shown on the screen without working on them.